enhancements/storage/csi-inline-vol-security.md

```diff
@@ -282,92 +314,58 @@ N/A
   and add them as reviewers to this enhancement.
+-->
 
 #### Support Procedures
 
-TODO
+**TODO**
 
 ## Implementation History
 
 - 2022-01-12: Initial draft
-
-## Drawbacks
-
-`CSIDrivers` that provide the `Ephemeral` volume capability should be able to
-provision these volumes in a secure manner. There is some concern in upstream
-Kubernetes that this plugin would give license for CSIDriver implementers to
-allow risky behaviors, increasing the attack surface of Kubernetes as a whole.
-
-This proposal adds a new admission plugin whose behavior and functionality is
-related to the existing PodSecurityAdmission plugin. The challenge of this
-approach is the handling of defaults when the CSI driver does not have an
-effective profile label, the namespace does not have a pod security enforcement
-label, or both. The admission plugin would need its own configuration for
-default behavior, and skew between its behavior and the PodSecurityAdmission
-admission plugin could lead to undesirable outcomes. For example,
-PodSecurityAdmission could be configured to enforce the “restricted” profile by
-default in all namespaces, whereas this new admission plugin could be
-configured to assume the `privileged` namespace profile is enforced by default.
-This could cause the admission plugin to allow admission of a pod that uses a
-`privileged` CSI driver in a namespace that has the “restricted” profile
-enforced.
+- 2022-09-08: Updates to original proposal and use webhook
 
 ## Alternatives
 
-### Augment the PodSecurityAdmission Plugin
-
-Rather than creating a new admission plugin, this capability could be added to
-the extending the PodSecurityAdmission plugin, this admission logic could be
-implemented in a separate validating admission admission plugin. A similar
-labeling scheme could be applied, and the admission plugin would admit or deny
-a pod based on the namespace’s pod security labels and the CSI driver’s
-effective pod security profile.
-
-This would greatly extend the scope of PodSecurityAdmission plugin beyond the
-original pod security standards. The standards were explicitly designed to
-allow additional admission controls for CSI ephemeral volumes to be built on
-top of the PodSecurityAdmission plugin. Extending it in this way is a
-non-starter.
-
-### Implement as a Validating Webhook
+### Implement as an admission plugin
 
-The plugin could be implemented as a validating webhook, which would eliminate
-the need to carry the plugin as a patch to Kubernetes proper. A separate
-webhook would incur extra network hops on every pod creation request. This
-would have significant performance impact on OpenShift - thus an admission
-plugin invoked directly on the kube-apiserver is appropriate.
+This was originally proposed as an admission plugin, rather than a webhook.
+However, this would require a carry patch in `openshift/kubernetes` which
```

Comment:
For the record, the carry patch could be quite clean, as there already is a place that registers OCP specific admission plugins [here](https://github.com/openshift/kubernetes/blob/ebabf6d0f11bae177af8edb131e9052c31546489/openshift-kube-apiserver/admission/admissionenablement/register.go#L25) and the admission plugins are then implemented in a separate repo (e.g. [scc](https://github.com/openshift/apiserver-library-go/tree/master/pkg/securitycontextconstraints/sccadmission)).

Reply:
This is a tough choice, but given that this is
1. required for security
2. is required to provide basic cluster functionality (this isn't a fancy add-on, it's just csi volumes)
3. must evaluate all pods as webhook because there is no mechanism to filter by "does this pod matter"

I think we should add it to openshift/kubernetes.  Look for where we add SCC.  Please add inspection of known "podspec-y objects" like deployments, deploymentconfigs, jobs, etc.  We can debate fail or warn in the PR or enhancement, but we have a fairly stable list now.

Reply:
Thanks for the inputs, changed it back to admission plugin.